include/wl/j_moves.h

```diff
@@ -26,6 +26,20 @@ class JackMove {
 
 	constexpr MapNode::ID destination() const noexcept { return where_to_; }
 	constexpr std::optional<JackResource> cost() const noexcept { return resource_; }
+
+	constexpr bool operator==(const JackMove& other) const noexcept {
+		return (where_to_ == other.where_to_) &&
+		       (resource_ == other.resource_);
+	}
```

Comment:
This defines the result of the expression `move1 == move2` where both variables have the type `wl::JackMove`.

We're just passing the buck down to comparing each of its member types (`wl::MapNode::ID` and `std::optional<wl::JackResource>`)